api/v2/routes_lens.go

```diff
@@ -28,59 +27,59 @@ func (api *API) submitIndexRequest(c *gin.Context) {
 			Fail(c, err)
 			return
 		}
+		indexType = pb.IndexReq_IPLD
 	default:
 		Fail(c, errors.New(eh.InvalidObjectTypeError))
 		return
 	}
-	// send the index request
-	resp, err := api.lens.Index(context.Background(), &request.Index{
-		Type:       forms["object_type"],
-		Identifier: forms["object_identifier"],
-		// allow for optional reindexing
-		Reindex: c.PostForm("reindex") == "yes",
+
+	resp, err := api.lens.Index(context.Background(), &pb.IndexReq{
+		Type:    indexType,
+		Hash:    forms["object_identifier"],
+		Options: &pb.IndexReq_Options{
+			//	Reindex: c.PostForm("reindex") == "true",
+		},
 	})
 	if err != nil {
 		api.LogError(c, err, eh.FailedToIndexError)(http.StatusBadRequest)
 		return
 	}
-	// return
-	Respond(c, http.StatusOK, gin.H{
-		"response": gin.H{
-			"lens_id":  resp.GetId(),
-			"keywords": resp.GetKeywords(),
-		},
-	})
+	Respond(c, http.StatusOK, gin.H{"response": resp.GetDoc()})
 }
 
-// submitSearchRequest is used to send a search request to lens
 func (api *API) submitSearchRequest(c *gin.Context) {
-	// extract key words to search for
-	keywords, exists := c.GetPostFormArray("keywords")
+	// query we will use to perform as the main search
+	query, exists := c.GetPostForm("query")
 	if !exists {
-		FailWithMissingField(c, "keywords")
+		FailWithMissingField(c, "query")
 		return
 	}
-	// for all keywords to lower-case
-	var keywordsLower []string
-	for _, word := range keywords {
-		keywordsLower = append(keywordsLower, strings.ToLower(word))
-	}
-	// send search query
-	resp, err := api.lens.Search(context.Background(), &request.Search{
-		Keywords: keywordsLower,
+
+	tags, _ := c.GetPostFormArray("tags")
+	categories, _ := c.GetPostFormArray("categories")
+	mimeTypes, _ := c.GetPostFormArray("mime_types")
+	hashes, _ := c.GetPostFormArray("hashes")
+	required, _ := c.GetPostFormArray("required")
```

Comment:
might be cleaner to declare these in a `var` block:

```go
	var (
		tags, _       = c.GetPostFormArray("tags")
		categories, _ = c.GetPostFormArray("categories")
		mimeTypes, _  = c.GetPostFormArray("mime_types")
		hashes, _     = c.GetPostFormArray("hashes")
		required, _   = c.GetPostFormArray("required")
	)
```

Reply:
good suggestion